src/main/kotlin/io/slama/events/autoroles.kt

```diff
@@ -55,12 +57,16 @@ class AutoRole(
     }
 
     override fun onButtonClick(event: ButtonClickEvent) {
-        if (!event.componentId.startsWith(name)) return
+        val matchResult = buttonIdRegex.find(event.componentId) ?: return
+
+        val name = matchResult.groupValues[1]
```

Comment:
use MatchResult::destructured (property) to expect clear groups

```kt
val (name, index) = matchResult.destructured.let { name, index -> name to index.toInt() }
```

this will produce a much better stacktrace if something goes wrong at some point

Reply:
Oh! I found the `destructured` thing but didn't find a nice solution to convert second group to Integer.